interview-questions.md

```diff
@@ -8,50 +8,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: The Javascript built-in method .map() takes the predefined parameters of value, index, and array. The parameter value is required; index and array are optional. 
+
+Researched answer: The Javascript biult-in method .map() takes in the predefined parameters of the current value of the array (value), the index of the array are the current iteration, and the array that the higher-order function is called on (array). The names of the parameters can technically be any name, but the order matters and will always be the same. It must take in at least one value which is required, but the index and array parameters are optional. 
 
-Researched answer:
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: the difference between map and filter is that map performs an iteration over an array and returns an array of the same amount of data, performing an action on each of the data included. Filter can be used when you want to return an array with only certain specifications, such as an array with only odd numbers. 
 
-Researched answer:
+Researched answer: The .map() method performs an iteration through an array and returns an array of the same length, or same amount of data. Depending on what we want to do with each of the data, .map() iterates through a certain array and returns a new array with new updated values. .map() does something to each value in the array. 
+The .filter() method performs an iteration through an array and returns a new array with only the values that satisfies a certain condition or specifications, therefore, filter needs an evaluation to in order to perform (ex. filtering through an array and returning an array with only odd numbers). The evaluation can be anything that returns a Boolean value or a truthy/falsey value. .filter() makes a decision about each item in the array and returns an array based on whatever condition is given. 
 
-1. What is iteration?
 
-Your answer:
+4. What is iteration?
 
-Researched answer:
+Your answer: Iteration is going through something (usually an array) and performing a function on it. It is doing something over and over again until it satisfies a condition. 
+
+Researched answer: Iteration is performing a block of code a certain amount of times or over and over again until it satifies a condition. Data types with length properties, such as strings and arrays, is what iterations can be performed on. Examples of iterations are for loops and higher-order functions. These functions performs an action on each item until a condition is met. A for loop in particular, must define a starting location, a condition to satisfy that will end the loop, and what action will take place during the iteration. 
```

Comment:
This answer is awesome! Straight to the point, then brought it home, and then really slammed it with the example! Nice 🙌🏽